### PR TITLE
Add tsc watch script to contract package.json

### LINKED ISSUE #308: Add `ts-watch` to `project-ts` template
### Describe the problem

Devs need a watch to watch and re-build TypeScript files for their smart contract automatically. 

### Describe the proposed solution

[`ts-watch`](https://www.npmjs.com/package/ts-watch)  allows them to do this. I propose installing this to devDependencies and then adding an npm script command (likely `buildw`) similar to our build command, but executing the `ts-watch` command.

### Alternatives considered

_No response_

### Importance

would make my life easier

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by ymekuria**
Closes #308 

This PR adds a tsc watch script to the contract package.json.

### CODE REVIEW

File: src/lib/project.js [hunk 1]
```diff
@@ -142,6 +142,12 @@ async function project({ name, ui }) {
   // `/dev/null` on Mac or Linux and 'NUL' on Windows is the only way to silence
   // Husky's install log msg. (Note: The contract project template commits
   // package-lock.json so we can use `npm ci` for faster installation.)
+
+  // Add a ts-watch script to the contract package.json
+  let contractPkgJson = fs.readJSONSync('package.json');
+  contractPkgJson.scripts['buildw'] = 'tsc -p tsconfig.json --watch';
+  fs.writeJSONSync('package.json', contractPkgJson, { spaces: 2 });
```
Comment (mitschabaude) [inline src/lib/project.js:149]: any reason why we don't put this script in the template `package.json` directly?

**Review by mitschabaude — COMMENTED** [review]
one question:

File: src/lib/project.js (hunk 1 above)
Comment (ymekuria) [reply src/lib/project.js]: Good call. I removed this and added it directly to the template.

**Review by mitschabaude — APPROVED** [review]
LGTM!

**Pull request merged**